201-week-3/app.js

```diff
@@ -74,28 +74,31 @@ Product.prototype.render_as_img = function(div1, h2_el1, pics1){
     pics1.src = this.url;
     h2_el1.textContent = this.name;   
 };
-
-new Product('bag', 'bag.jpg');
-new Product('banana', 'banana.jpg');
-new Product('bathroom', 'bathroom.jpg');
-new Product('boots', 'boots.jpg');
-new Product('breakfast', 'breakfast.jpg')
-new Product('bubblegum', 'bubblegum.jpg');
-new Product('chair', 'chair.jpg');
-new Product('cthulhu', 'cthulhu.jpg');
-new Product('dog-duck', 'dog-duck.jpg');
-new Product('dragon', 'dragon.jpg');
-new Product('pen', 'pen.jpg');
-new Product('pet-sweep', 'pet-sweep.jpg');
-new Product('scissors', 'scissors.jpg');
-new Product('shark', 'shark.jpg');
-new Product('sweep', 'sweep.png');
-new Product('tauntaun', 'tauntaun.jpg');
-new Product('unicorn', 'unicorn.jpg');
-new Product('usb', 'usb.gif');
-new Product('water-can', 'water-can.jpg');
-new Product('wine-glass', 'wine-glass.jpg');
-
+if(localStorage.getItem(stringyProducts) === null){

```

Comment:
this is breaking because you are checking to see if there is something called 'undefined' in local storage
localStorage.getItem takes in a string as its argument
the key that you are setting in local storage is called 'Products'
that is the name you need to use to retrieve your data